assets/prototype/application/ui/components/display/entityListFilterBar/useEntityListFilterState.js

```diff
@@ -0,0 +1,102 @@
+import { useReducer, useEffect } from '@wordpress/element';
+import { assocPath, path } from 'ramda';
+
+const useEntityListFilterState = (listId) => {
+	const initialState = {
+		[listId]: {
+			perPage: 6,
+			searchText: '',
+			showEntityFilters: false,
+			view: 'list',
+		},
+	};
+	const [state, dispatch] = useReducer(reducer, initialState);
+
+	useEffect(() => {
+		console.log('useEntityListFilterState >>>', state);
+	}, [state]);
+
+	const setGridView = () => {
+		dispatch({
+			listId,
+			type: 'SET_VIEW',
+			view: 'grid',
+		});
+	};
+
+	const setListView = () => {
+		dispatch({
+			listId,
+			type: 'SET_VIEW',
+			view: 'list',
+		});
+	};
+
+	const setPerPage = (perPage) => {
+		dispatch({
+			listId,
+			perPage,
+			type: 'SET_PER_PAGE',
+		});
+	};
+
+	const setSearchText = (searchText) => {
+		dispatch({
+			listId,
+			searchText,
+			type: 'SET_SEARCH_TEXT',
+		});
+	};
+
+	const toggleEntityFilters = () => {
+		const showEntityFilters = !state[listId].showEntityFilters;
+
+		dispatch({
+			listId,
+			type: 'TOGGLE_ENTITY_FILTERS',
+			showEntityFilters,
+		});
+	};
+
+	const getStateProp = (prop) => path([listId, prop], state);
```

Comment:
I've named this simply because the fact that it is inside the `useEntityListFilterState.js` explains the context it is being used.